subsys/net/lib/http/http_server_http1.c

```diff
@@ -489,7 +489,10 @@ int handle_http1_request(struct http_client_ctx *client)
 		char *needed_upgrade = "h2c\r\n";
 
 		if (client->websocket_upgrade) {
-			if (IS_ENABLED(CONFIG_WEBSOCKET)) {
+			if (IS_ENABLED(CONFIG_HTTP_SERVER_WEBSOCKET)) {
+				client->current_detail =
+					get_resource_detail(client->url_buffer,
+							    &path_len, true);
```

Comment:
We don't seem to handle well the situation when no websocket resource has been found. No check is here, and `handle_http1_to_websocket_upgrade()` seems to dereference `client->current_detail` unconditionally.

Personally, I'd also reconsider whether there's a need to modify `get_resource_detail()`.  `handle_http1_to_websocket_upgrade()` could simply check the resource type and report an error if the resource is of a different type. That way we keep the websocket processing details withing the websocket part of the server implementation.